codec-http2/src/main/java/io/netty/handler/codec/http2/HpackEncoder.java

```diff
@@ -380,6 +392,24 @@ private int getIndex(CharSequence name) {
         return -1;
     }
 
+    /**
+     * Returns the lowest index value for the header field name in the dynamic table. Returns -1 if
+     * the header field name is not in the dynamic table.
+     */
+    private int getIndexInsensitive(CharSequence name) {
```

Comment:
How much does this buy us? I'd expect not much. Since we only pay the equals check if there is a hash collision, and these are header names which are short, it seems like there's not room for gain here.

Reply:
Maybe, but it's about code clarity.  This method says it doesn't care about security, as opposed to `getIndexSensitive`.   

Also in the case the header names change frequently, I'd expect there would be frequent collisions and (evictions).

Reply:
@ejona86 WDYT ?

Reply:
To me the sensitive/nonsensitive split just is a bit more confusing. I'd have preferred dumping a comment on the previous `getIndex`: `// use constant-time comparison since used with sensitive headers`. And for the value look ups `// not used for sensitive headers, so no need to use constant-time comparison`.

If we really like the split names, then I'd just have `getIndexInsensitive()` call `getIndexSensitive()` in the implementation.

Header name changes doesn't have anything to do with frequency of collisions. Only the number of entries plays a part in collision rate. I still maintain that for header names the difference between constant- vs variable-time comparison is well below what we should care about. I'd much rather optimize for code clarity and size here by not duplicating the getIndex implementation.